existential-vs-universal-quantification/article.md

```diff
@@ -0,0 +1,295 @@
+# Existential and Universal Quantification in Haskell
+
+In this article I give a brief introduction into one of the most common Haskell extensions – `ExplicitForAll` – and syntax constructions that this extension brings to language.
+It is used in about all big haskell projects, so it is very nice to either become acquainted with it or to refresh it in your memory if you already know about it.
+
+Stay tuned if you want to know how to create lists of differently typed variables, how to add complex constraints, and more!
+
+## Introduction
+
+By default, quantification in Haskell is implicit.
+Without the `ExplicitForAll` extension, you can't write quantifiers. Moreover, all variables are universally quantified, and existential quantification is off.
+The key point of both universal and existential quantification is choosing where, we are going to instantiate a type variable - where it is defined, or where it is used.
+
+## Universal quantification and `ExplicitForAll`
+
+By default, all type variables are considered universally quantified.
+This means that every time you use a function like `id`, you can _instantiate_ the universally quantified type variable `a` to (i.e., replace it with) any type.
+
+### Usage
+
+Let's try to add `forall` manually.
```

Comment:
`forall` is not defined yet, it will help if you show `forall` in the intro and talk about what it is just a bit. This article assumes a lot of Haskell competence.

Reply:
Agree, if you explain what is a quanitifier correctly, you will be able to say that we are adding a quantifier and that's all.